### PR TITLE
Add child collection to the entity class in Firestore sample

### LINKED ISSUE #2387: Add child collection to the entity class in Firestore sample
Currently we only have fields of simple types. We should demonstrate collection usage as well.

### PR DESCRIPTION
**Pull request by dmitry-s**
fixes #2387

### CODE REVIEW

File: spring-cloud-gcp-samples/spring-cloud-gcp-data-firestore-sample/src/test/java/com/example/FirestoreSampleApplicationTests.java [hunk 1]
```diff
@@ -70,9 +70,9 @@ public void saveUserTest() {
 		User[] users = restTemplate.getForObject("/users", User[].class);
 		assertThat(users).isEmpty();
 
-		sendPostRequestForUser("Alpha", 49);
-		sendPostRequestForUser("Beta", 23);
-		sendPostRequestForUser("Delta", 49);
+		sendPostRequestForUser("Alpha", 49, "rat-Snowflake");
+		sendPostRequestForUser("Beta", 23, "");
+		sendPostRequestForUser("Delta", 49, "fish-Dory,spider-Man");
```
Comment (meltsufin) [inline spring-cloud-gcp-samples/spring-cloud-gcp-data-firestore-sample/src/test/java/com/example/FirestoreSampleApplicationTests.java:75]: Can we add assertions for verifying that these pets were added?

File: spring-cloud-gcp-samples/spring-cloud-gcp-data-firestore-sample/src/main/resources/static/index.html [hunk 2]
```diff
@@ -37,16 +37,18 @@ <h2>Firestore Control Panel</h2>
...
-    </form>
-  </div>
+    <div class="panel">
+        <b>Create and save a new User</b>
+        <form action="/users/saveUser" method="post">
+            Name: <input type="text" name="name" value="Joe Smith">
+            <br>
+            Age: <input type="text" name="age" value="44">
+            <br>
+            Pets (type-name, coma separated): <input type="text" name="pets" value="dog-Dusty,cat-Tosha">
```
Comment (dzou) [inline spring-cloud-gcp-samples/spring-cloud-gcp-data-firestore-sample/src/main/resources/static/index.html:47]:
```suggestion
            Pets (comma-separated values {pet-type}-{name} for each pet, such as "fish-dory"): <input type="text" name="pets" value="dog-Dusty,cat-Tosha">
```

File: spring-cloud-gcp-samples/spring-cloud-gcp-data-firestore-sample/src/main/java/com/example/User.java [hunk 3]
```diff
@@ -33,12 +37,16 @@
 
 	int age;
 
+	List<Pet> pets;
```
Comment (dzou) [inline spring-cloud-gcp-samples/spring-cloud-gcp-data-firestore-sample/src/main/java/com/example/User.java:40]: Might be helpful to add an extra comment here to draw attention to it, like:

```
// This represents a Firestore child collection
List<Pet> pets;
```

Comment (dmitry-s) [reply spring-cloud-gcp-samples/spring-cloud-gcp-data-firestore-sample/src/main/java/com/example/User.java:40]: I think that should be clear from the context.

**Pull request merged**